packages/auth/src/providers/cognito/apis/confirmSignUp.ts

```
@@ -32,8 +32,7 @@ export async function confirmSignUp(
 ): Promise<AuthSignUpResult<AuthStandardAttributeKey | CustomAttribute>> {
 	const { username, confirmationCode, options } = confirmSignUpRequest;
 
-	// TODO: replace by singleton implementation.
-	const config = Amplify.config;
+	const authConfig = AmplifyV6.getConfig().Auth;
```

Again probably out of scope for this PR, but if Auth hasn't been configured should we throw an error? Maybe we can have a little helper function to do this across APIs (with a similar pattern for other categories when we get there).

```
export const getAuthConfig(amplifySingleton) {
  const authConfig = AmplifyV6.getConfig().Auth;
  
  if (!authConfig) {
    throw new AmplifyError(...);
  }
  
  return authConfig;
}
```

---

I like that idea, the thing is that in the end if we support different service implementation, we cannot enforce it on the Singleton, it should be on the implementation that is consuming the config

---

The example code I posted would live with the implementation, it would just get called in each API that requires the config.